src/gen/lib/AST_grammar_conv.ml

```diff
@@ -123,6 +130,13 @@ let tsort_rules find_rule_aliases rules =
       { name; aliases; is_rec; body }) group
   ) sorted
 
+let filter_extras bodies =
+  List.filter_map (fun (x : Tree_sitter_t.rule_body) ->
+    match x with
+    | SYMBOL name -> Some name
+    | _ -> None
```

Comment:
You can have something other than a SYMBOL in extras?

Reply:
The cases I'm aware of are:

* regexps (comments, whitespace): those do not show up in the json output.
* symbols: those show up in the parsing result, and need to be removed.

I need to check what happens for string constants.

There could also be a construct like a seq(). Since they don't have a name, I presume they don't show up in the output.

Reply:
I thought STRING constants could be supported in the list of "extras" but they cause a runtime error to the tree-sitter parser. The doc says they must be tokens, and we only care if they have a name, because otherwise they don't show up in the parser's output and there's nothing for us to do.